File: JAVA/Clients/clients/src/main/java/com/futureprograms/clients/util/ValidationHelper.java
package com.futureprograms.clients.util;

import lombok.extern.slf4j.Slf4j;
import org.springframework.web.multipart.MultipartFile;

/**
 * Utilidad de validación centralizada
 * Sigue el principio DRY (Don't Repeat Yourself)
 */
@Slf4j
public final class ValidationHelper {

    private static final long MAX_IMAGE_SIZE = 20 * 1024 * 1024; // 20MB

    private ValidationHelper() {
        throw new AssertionError("No se puede instanciar esta clase");
    }

    /**
     * Valida que un archivo de imagen sea válido
     */
    public static void validateImageFile(MultipartFile file) {
        if (file == null || file.isEmpty()) {
            throw new IllegalArgumentException(ApiConstants.ERR_NO_IMAGE);
        }

        if (file.getSize() > MAX_IMAGE_SIZE) {
            throw new IllegalArgumentException("El archivo excede el tamaño máximo permitido");
        }

        String contentType = file.getContentType();
        if (!isValidImageType(contentType)) {
            throw new IllegalArgumentException(ApiConstants.ERR_IMAGE_TYPE_INVALID);
        }
    }

    /**
     * Valida que el tipo de contenido sea una imagen válida
     */
    public static boolean isValidImageType(String contentType) {
        if (contentType == null) {
            return false;
        }
        return contentType.equals("image/jpeg")
                || contentType.equals("image/png")
                || contentType.equals("image/gif")
                || contentType.equals("image/webp");
    }

    /**
     * Valida que un string sea válido (no null ni vacío)
     */
    public static boolean isValidString(String value) {
        return value != null && !value.trim().isEmpty();
    }

    /**
     * Valida que una ruta de archivo sea segura (sin path traversal)
     */
    public static boolean isValidFilePath(String path) {
        if (path == null || path.trim().isEmpty()) {
            return false;
        }
        return !path.contains("..") && !path.contains("//") && path.matches("^[a-zA-Z0-9/_.-]+$");
    }

    /**
     * Extrae la extensión de un archivo
     */
    public static String getFileExtension(String filename) {
        if (filename == null || !filename.contains(".")) {
            return "jpg";
        }
        return filename.substring(filename.lastIndexOf(".") + 1).toLowerCase();
    }

    /**
     * Valida email básico
     */
    public static boolean isValidEmail(String email) {
        return isValidString(email) && email.matches("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$");
    }

    /**
     * Valida que una contraseña sea segura (mínimo 8 caracteres)
     */
    public static boolean isValidPassword(String password) {
        return isValidString(password) && password.length() >= 8;
    }
}

File: ASP .NET/Clients/Helpers/ValidationHelper.cs
namespace Clients.Helpers;

/// <summary>
/// Utilidad de validación centralizada
/// Sigue el principio DRY (Don't Repeat Yourself)
/// Equivalente a ValidationHelper.java en la API JAVA
/// </summary>
public static class ValidationHelper
{
    private const long MaxImageSize = 20 * 1024 * 1024; // 20MB

    /// <summary>
    /// Valida que un archivo de imagen sea válido
    /// </summary>
    public static void ValidateImageFile(IFormFile file)
    {
        if (file == null || file.Length == 0)
        {
            throw new InvalidOperationException("Por favor selecciona una imagen");
        }

        if (file.Length > MaxImageSize)
        {
            throw new InvalidOperationException("El archivo excede el tamaño máximo permitido (20MB)");
        }

        if (!IsValidImageType(file.ContentType))
        {
            throw new InvalidOperationException("Tipo de archivo no permitido. Solo se permiten imágenes (jpg, png, gif, webp)");
        }
    }

    /// <summary>
    /// Valida que el tipo de contenido sea una imagen válida
    /// </summary>
    public static bool IsValidImageType(string? contentType)
    {
        if (string.IsNullOrEmpty(contentType))
        {
            return false;
        }

        return contentType == "image/jpeg"
            || contentType == "image/png"
            || contentType == "image/gif"
            || contentType == "image/webp";
    }

    /// <summary>
    /// Valida que un string sea válido (no null ni vacío)
    /// </summary>
    public static bool IsValidString(string? value)
    {
        return !string.IsNullOrWhiteSpace(value);
    }

    /// <summary>
    /// Valida que una ruta de archivo sea segura (sin path traversal)
    /// </summary>
    public static bool IsValidFilePath(string? path)
    {
        if (string.IsNullOrWhiteSpace(path))
        {
            return false;
        }

        return !path.Contains("..")
            && !path.Contains("//")
            && System.Text.RegularExpressions.Regex.IsMatch(path, @"^[a-zA-Z0-9/_.-]+$");
    }

    /// <summary>
    /// Extrae la extensión de un archivo
    /// </summary>
    public static string GetFileExtension(string? filename)
    {
        if (string.IsNullOrEmpty(filename) || !filename.Contains("."))
        {
            return "jpg";
        }

        return filename.Substring(filename.LastIndexOf(".") + 1).ToLowerInvariant();
    }

    /// <summary>
    /// Valida email básico
    /// </summary>
    public static bool IsValidEmail(string? email)
    {
        if (!IsValidString(email))
        {
            return false;
        }

        return System.Text.RegularExpressions.Regex.IsMatch(
            email!,
            @"^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$");
    }

    /// <summary>
    /// Valida que una contraseña sea segura (mínimo 8 caracteres)
    /// </summary>
    public static bool IsValidPassword(string? password)
    {
        return IsValidString(password) && password!.Length >= 8;
    }
}
